crates/db_views/src/post/post_view.rs

```diff
@@ -459,94 +368,79 @@ impl<'a> PostQuery<'a> {
-            query = query.filter(post::name.not_ilike(pattern.clone()));
-            query = query.filter(post::url.is_null().or(post::url.not_ilike(pattern.clone())));
-            query = query.filter(
-              post::body
-                .is_null()
-                .or(post::body.not_ilike(pattern.clone())),
-            );
-          }
+
+      if let Some(keyword_blocks) = o.keyword_blocks {
+        for keyword in keyword_blocks {
+          let pattern = format!("%{}%", keyword);
+          query = query.filter(post::name.not_ilike(pattern.clone()));
+          query = query.filter(post::url.is_null().or(post::url.not_ilike(pattern.clone())));
+          query = query.filter(
+            post::body
+              .is_null()
+              .or(post::body.not_ilike(pattern.clone())),
+          );
         }
       }
     }
 
-    let (limit, offset) = limit_and_offset(o.page, o.limit)?;
-    query = query.limit(limit).offset(offset);
+    // Filter by the time range
+    if let Some(time_range_seconds) = o.time_range_seconds {
+      query = query.filter(post::published.gt(now() - seconds_to_pg_interval(time_range_seconds)));
+    }
 
-    let query = if o.read_only.unwrap_or_default() {
-      paginate(
-        query,
-        o.page_after.map(|c| c.post_actions),
-        None,
-        o.page_back.unwrap_or_default(),
-      )
-      .filter(post_actions::read.is_not_null())
-      .then_desc(post_actions_keys::read)
-      .as_query()
-    } else {
-      let mut query = paginate(
-        query,
-        o.page_after.map(|c| c.post),
-        o.page_before_or_equal,
-        o.page_back.unwrap_or_default(),
-      );
+    // Only sort by ascending for Old
+    let sort = o.sort.unwrap_or(Hot);
+    let sort_direction = asc_if(sort == Old);
 
-      // featured posts first
-      query = if o.community_id.is_none() || o.community_id_just_for_prefetch {
-        query.then_desc(key::featured_local)
-      } else {
-        query.then_desc(key::featured_community)
-      };
+    let mut pq = paginate(
+      query,
+      sort_direction,
+      o.cursor_data,
+      o.cursor_before_data,
+      o.page_back,
+    );
 
-      // then use the main sort
-      query = match o.sort.unwrap_or(Hot) {
-        Active => query.then_desc(key::hot_rank_active),
-        Hot => query.then_desc(key::hot_rank),
-        Scaled => query.then_desc(key::scaled_rank),
-        Controversial => query.then_desc(key::controversy_rank),
-        New => query.then_desc(key::published),
-        Old => query.then_desc(ReverseTimestampKey(key::published)),
-        NewComments => query.then_desc(key::newest_comment_time),
-        MostComments => query.then_desc(key::comments),
-        Top => query.then_desc(key::score),
-      };
+    // featured posts first
+    pq = if o.community_id.is_none() {
+      pq.then_order_by(key::featured_local)
+    } else {
+      pq.then_order_by(key::featured_community)
+    };
```

Comment:
these will not work correctly if `sort == Old`, and this can be fixed using the same solution used in comment view